backfill.py

```diff
@@ -130,27 +146,165 @@ def get_result_from_durations(start_up_durations_path, analyzed_path):
+               "\n\n {error}".format(br=branch, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def get_commits_for_date_range(startdate, enddate, repository_path):
+    commit_proc = subprocess.run(
+        ["git", "log", "--since={start}".format(start=startdate),
+         "--until={end}".format(end=enddate), "--pretty=format:'%H'"],
+        cwd=repository_path, capture_output=True, text=True).stdout
+
+    if commit_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit range: {start}..{end}" +
+               "The associated error message was:\n\n {error}".format(
+                start=startdate, end=enddate, error=commit_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+    commit_string = proc.replace("'", "")
+    return commit_string.split("\n")
+
+
+def get_all_commits_in_commits_range(start_commit, end_commit, repository_path):
+    commit_proc = subprocess.run(
+        ["git", "rev-list", "--ancestry-path", start_commit + ".." + end_commit],
+        cwd=repository_path, capture_output=True, text=True).stdout
+    commit_list = commit_proc.replace("'", "")
+
+    if commit_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit range: {start}..{end}" +
+               "The associated error message was:\n\n {error}".format(
+                start=start_commit, end=end_commit, error=commit_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+    return commit_list.split("\n")
+
+
+def build_apk_for_commit(hash, repository_path, build_type):
+    checkout_proc = subprocess.run(["git", "checkout", hash], cwd=repository_path, capture_output=True)
+    assemble_proc = subprocess.run(["./gradlew", "assemble"+build_type], cwd=repository_path, capture_output=True)
+
+    if checkout_proc.returncode != 0:
+        print(("\n\nSomething went wrong while checking out this commit: {commit} . The associated error message was:"
+               "\n\n {error}".format(commit=hash, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+    elif assemble_proc.returncode != 0:
+        print(("\n\nSomething went wrong while assembling this build: {build} . The associated error message was:"
+               "\n\n {error}".format(build=build_type, error=checkout_proc.stderr.decode('utf-8').strip("\n"))),
+              file=sys.stderr)
+
+
+def build_apk_path_string(repository_path, build_type, phone_architecture):
+    build_apk_destination = repository_path + "/app/build/outputs/apk/" + build_type + "/"
+    apk_name = "app-" + phone_architecture + "-" + build_type + ".apk"
+    return build_apk_destination + apk_name
+
+
+def move_apk_to_cwd(apk_path, commit_hash):
+    new_apk_name = "apk_commit_" + commit_hash + ".apk"
+    proc = subprocess.run(["mv", apk_path, new_apk_name])
```

Comment:
nitpick (blocking): `apk_path` will not exist when run using the `arm64-v8a-` argument. Here's the error I got:
```
mv: rename ../fenix/app/build/outputs/apk/nightly/app-arm64-v8a--nightly.apk to apk_commit_c55c252f9af2b91c5fef8ed47bb64b9390fbadbb.apk: No such file or directory
Traceback (most recent call last):
  File "/Users/mcomella/dev/perf-tools/backfill.py", line 314, in <module>
    main()
  File "/Users/mcomella/dev/perf-tools/backfill.py", line 294, in main
    array_of_apk_metadata = build_apks_for_commits(
  File "/Users/mcomella/dev/perf-tools/backfill.py", line 248, in build_apks_for_commits
    new_apk_name = move_apk_to_cwd(built_apk_name, commit)
  File "/Users/mcomella/dev/perf-tools/backfill.py", line 229, in move_apk_to_cwd
    "\n\n {error}".format(apk=apk_path, error=proc.stderr.decode('utf-8').strip("\n"))),
AttributeError: 'NoneType' object has no attribute 'decode'
```

This works for armv7 though. To keep things simpler at implementation time, I'd probably have hard-coded everything to be armv7 (like we did with download) and add this as a subsequent PR to avoid having a combinatorial explosion of test cases to test when I made a change to the script.